src/main/java/com/adobe/aem/compgenerator/AemCompGenerator.java

```diff
@@ -40,44 +72,151 @@
     private static final Logger LOG = LogManager.getLogger(AemCompGenerator.class);
 
     public static void main(String[] args) {
```

Comment:
Can we still require a path to a `data-config.json` file so that the file is not required to sit at the top level of the user's code project?  This is how the command line works as well, and it would also allow someone to maintain a set of config files for different components and simply point to the one they want to use.

This goes along with the comment that I feel that "ui" should be an explicit option rather than the default action, especially since it may not be intuitive that you use server mode by not specifying a config file (and the server mode still uses a config file)